images/gitlab/tests/main.tf

```diff
@@ -0,0 +1,98 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+variable "digests" {
+  description = "The image digest to run tests over."
+  type = object({
+    kas = string
+  })
+}
+
+variable "namespace" {
+  description = "The namespace to install Gitlab in."
+}
+
+// Invoke the image's version command.
+// $IMAGE_NAME is populated with the image name by digest.
+// TODO: Update or remove this test as appropriate.
+data "oci_exec_test" "kas-version" {
+  digest = var.digests.kas
+  script = "docker run --rm $IMAGE_NAME -v"
+}
+
+resource "random_pet" "suffix" {}
+
+locals {
+  namespace = "gitlab-${random_pet.suffix.id}"
+}
+
+data "oci_string" "kas-ref" { input = var.digests.kas }
+
+resource "helm_release" "gitlab" {
+  name = "gitlab"
+
+  repository = "https://charts.gitlab.io"
+  chart      = "gitlab"
+
+  namespace        = local.namespace
+  create_namespace = true
+
+  timeout = 400
+
+  // Set to false to then wait for the specific images are deployed
+  wait = false
```

Comment:
I am adding a wait=false to mainly focus on the gitlab-kas image for the moment.